atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -1617,16 +1607,98 @@ public void getSentinelValuesStressTest() {
-                TABLE_SWEPT_CONSERVATIVE,
+                TABLE_SWEPT_THOROUGH, ImmutableMap.of(TEST_CELL, new byte[0]), Value.INVALID_VALUE_TIMESTAMP);
+
+        assertThat(keyValueService.get(TABLE_SWEPT_THOROUGH, ImmutableMap.of(TEST_CELL, 0L)))
+                .isNotEmpty();
+
+        Transaction t1 = txManager.createNewTransaction();
+        assertThatCode(() -> t1.get(TABLE_SWEPT_THOROUGH, ImmutableSet.of(TEST_CELL)))
+                .doesNotThrowAnyException();
+
+        assertThat(keyValueService.get(TABLE_SWEPT_THOROUGH, ImmutableMap.of(TEST_CELL, 0L)))
+                .isEmpty();
+    }
+
+    @Test
+    public void orphanedSentinelsCoveredWithUncommitedGetDeletedAfterDetectionInThoroughlySweptTable() {
+        writeSentinelToTestTable(TABLE_SWEPT_THOROUGH, TEST_CELL);
+        putUncommittedAtFreshTimestamp(TABLE_SWEPT_THOROUGH, TEST_CELL);
+
+        Transaction t1 = txManager.createNewTransaction();
+        assertThatCode(() -> t1.get(TABLE_SWEPT_THOROUGH, ImmutableSet.of(TEST_CELL)))
+                .doesNotThrowAnyException();
+
+        assertThat(keyValueService.get(TABLE_SWEPT_THOROUGH, ImmutableMap.of(TEST_CELL, 0L)))
+                .isEmpty();
+    }
+
+    @Test
+    public void nonOrphanedSentinelsDoNotGetDeletedAfterDetectionInThoroughlySweptTable() {
+        Transaction t1 = txManager.createNewTransaction();
+        writeSentinelToTestTable(TABLE_SWEPT_THOROUGH, TEST_CELL);
+
+        commitWrite(TABLE_SWEPT_THOROUGH, TEST_CELL);
+
+        assertThatThrownBy(() -> t1.get(TABLE_SWEPT_THOROUGH, ImmutableSet.of(TEST_CELL)))
+                .isInstanceOf(TransactionFailedRetriableException.class)
+                .hasMessageContaining("Tried to read a value that has been deleted.");
+
+        assertThat(keyValueService.get(TABLE_SWEPT_THOROUGH, ImmutableMap.of(TEST_CELL, 0L)))
+                .isNotEmpty();
+    }
+
+    @Test
+    public void testSentinelsDeletionForDifferentCasesInThoroughTable() {
+        Transaction t1 = txManager.createNewTransaction();
+
+        Cell testCell3 = Cell.create(PtBytes.toBytes("super"), PtBytes.toBytes("ez"));
+
+        writeSentinelToTestTable(TABLE_SWEPT_THOROUGH, TEST_CELL);
+        writeSentinelToTestTable(TABLE_SWEPT_THOROUGH, TEST_CELL_2);
+        writeSentinelToTestTable(TABLE_SWEPT_THOROUGH, testCell3);
+
+        commitWrite(TABLE_SWEPT_THOROUGH, TEST_CELL);
+        putUncommittedAtFreshTimestamp(TABLE_SWEPT_THOROUGH, TEST_CELL_2);
+
+        assertThatThrownBy(() -> t1.get(TABLE_SWEPT_THOROUGH, ImmutableSet.of(TEST_CELL)))
+                .isInstanceOf(TransactionFailedRetriableException.class)
+                .hasMessageContaining("Tried to read a value that has been deleted.");
+
+        assertThat(keyValueService.get(TABLE_SWEPT_THOROUGH, ImmutableMap.of(TEST_CELL, 0L)))
+                .containsExactlyEntriesOf(
+                        ImmutableMap.of(TEST_CELL, Value.create(new byte[0], Value.INVALID_VALUE_TIMESTAMP)));
+    }
```

Comment:
Did we want to do something about test cells 2 and 3 after writing them? I think you want to read everything on line 1667, and then check the others are still there

Reply:
oops